web/src/utils/columns.ts

```diff
@@ -47,213 +49,137 @@ export enum FilterType {
-    }
-    case ColumnsId.dstport: {
-      return flow.value.IPFIX.DstPort;
-    }
-    case ColumnsId.proto: {
-      return flow.value.IPFIX.Proto;
-    }
-    case ColumnsId.bytes: {
-      return flow.value.IPFIX.Bytes;
-    }
-    case ColumnsId.packets: {
-      return flow.value.IPFIX.Packets;
-    }
+const sortNumbers = (a: number, b: number) => {
+  if (!isNaN(a) && !isNaN(b)) {
+    return a - b;
+  } else if (!isNaN(a)) {
+    return 1;
   }
+  return -1;
 };
 
-export const getSortFunctionFromColumnId = (colId: ColumnsId) => {
-  switch (colId) {
-    case ColumnsId.srcport:
-    case ColumnsId.dstport: {
-      return (a: ParsedStream, b: ParsedStream, isDesc: boolean) => {
-        return comparePort(
-          getFlowValueFromColumnId(isDesc ? a : b, colId),
-          getFlowValueFromColumnId(isDesc ? b : a, colId)
-        );
-      };
-    }
-    case ColumnsId.srcaddr:
-    case ColumnsId.dstaddr: {
-      return (a: ParsedStream, b: ParsedStream, isDesc: boolean) => {
-        return ipCompare(
-          getFlowValueFromColumnId(isDesc ? a : b, colId) as string,
-          getFlowValueFromColumnId(isDesc ? b : a, colId) as string
-        );
-      };
-    }
-    case ColumnsId.proto: {
-      return (a: ParsedStream, b: ParsedStream, isDesc: boolean) => {
-        const f1Name: string = protocols[getFlowValueFromColumnId(isDesc ? a : b, colId)]?.name;
-        const f2Name: string = protocols[getFlowValueFromColumnId(isDesc ? b : a, colId)]?.name;
-        if (f1Name && f2Name) {
-          return f1Name.localeCompare(f2Name);
-        } else if (f1Name) {
-          return 1;
-        } else {
-          return -1;
-        }
-      };
-    }
-    case ColumnsId.timestamp:
-    case ColumnsId.bytes:
-    case ColumnsId.packets: {
-      return (a: ParsedStream, b: ParsedStream, isDesc: boolean) => {
-        const f1Value = Number(getFlowValueFromColumnId(isDesc ? a : b, colId));
-        const f2Value = Number(getFlowValueFromColumnId(isDesc ? b : a, colId));
-        if (!isNaN(f1Value) && !isNaN(f2Value)) {
-          return f1Value - f2Value;
-        } else if (!isNaN(f1Value)) {
-          return 1;
-        } else {
-          return -1;
-        }
-      };
-    }
-    default: {
-      return (a: ParsedStream, b: ParsedStream, isDesc: boolean) => {
-        const f1Value = getFlowValueFromColumnId(isDesc ? a : b, colId) as string;
-        const f2Value = getFlowValueFromColumnId(isDesc ? b : a, colId) as string;
-        if (f1Value && f2Value) {
-          return f1Value.localeCompare(f2Value);
-        } else if (f1Value) {
-          return 1;
-        } else {
-          return -1;
-        }
-      };
-    }
+const sortStrings = (a: string, b: string) => {
+  if (a && b) {
+    return a.localeCompare(b);
+  } else if (a) {
+    return 1;
   }
+  return -1;
+};
+
+const sortProtocols = (a: number, b: number) => {
+  return sortStrings(protocols[a]?.name, protocols[b]?.name);
 };
```

Comment:
Should we move this to another file in `utils` folder ? 

We already have separate files for `ip` and `port`. Maybe we should make a `filter.ts` file and put everything inside ?

Reply:
good point i'll do that

Reply:
after looking at that, `port.ts` has more than just the comparison function, so I'd rather keep it and organize per data type (create `protocol.ts`)

Reply:
sure, thx